docs/src/integrate/patina_dxe_core_requirements.md

```diff
@@ -165,6 +165,42 @@ Resource Descriptor HOB v2 describing it.
 > Platforms must create Resource Descriptor HOB v2s for all memory resources including MMIO and reserved memory with
 > a valid cacheability attribute set.
 
+#### 2.2.1 Resource Descriptor Hob v2 Guidance
+
+Resource Descriptor HOBs are produced by a platform to describe ranges that compose the system's memory map.
+Historically, some platforms have only described memory that is in use, but the HOBs should describe the
+entire system memory map, as it stands, at the end of the HOB-producer phase. This includes caching
+attributes in V2 resource descriptor HOBs so that Patina is able to consume the data and provide enhanced
+memory protections from the start of DXE.
+
+##### PEI Video Region Example
+
+By the start of DXE, most system memory should be configured as Write Back (EFI_MEMORY_WB), and most Memory Mapped I/O
+should be configured as uncached (EFI_MEMORY_UC). There are exceptions to this, based upon platform level decisions.
+For example, if the pre-DXE phase configured a video device and displayed something on the screen, then the memory
+region associated with the frame buffer, for performance, should be reflected for that region of memory.
+
+##### Firmware Devices
+
+SPI flash accessiable through MMIO is another complex example.
+
+Consider the following scenario:
+
+- 64MB SPI part, mapped to 0xFC00_0000 - 0xFFFF_FFFF.
+- FVs in the SPI part
+  - 0xFD00_0000 - 0xFD04_FFFF - NVRAM
+  - 0xFD05_0000 - 0xFE40_FFFF - DXE code
+  - 0xFF00_0000 - 0xFFFF_FFFF - PEI code
+- HPET enabled, mapped to 0xFED0_0000 - 0xFED0_03FF.
+
+Note both the HPET region and the FV hobs overlaps with the SPI region.
+
+The HPET should have a MMIO resource descriptor HOB for its region.
+By the time end of PEI occurs, the FVs will have been shadowed to memory, so the FV resource descriptor HOBs should
+describe the FV regions in memory.
+
+The SPI flash can be reported around the HPET region as MMIO resources with the uncached attribute.
```

Comment:
I think that it is permissible to mark the whole flash region as MMIO resources with UC as the specified attribute in the HobList. While the varstore might be marked WP in PEI for read perf, nothing can touch the varstore in DXE until the necessary driver stack is up, and that driver stack has more information and context about what the caching attributes for the varstore should be and can apply them as part of the initialization of DXE var services.

I'm not saying one way or the other is preferable; I think either approach is viable. I more just want to note that it is a common pattern to describe only very coarse blocks at in the hoblist at DXE start and allow various drivers to carve it up and re-attribute it later.